### DISCUSSION TITLE
Allowing field to be undefined but not null

### DISCUSSION THREAD

**Discussion by jdc18** [dotansimha/graphql-code-generator#5358] [Q&A] [answered]
So I am having trouble making my resolvers work with my services/controllers that connect to my database with mongoose.
I am getting most of the time this,
```
Type 'string | null | undefined' is not assignable to type 'string | undefined'.
  Type 'null' is not assignable to type 'string | undefined'.
```
And I understand the error in Typescript but what if I don't want to the field to be nullable, just undefined.  Here for example
I have my this interface

```
interface IUpdateExperience {
    _id: IExperience['_id'];
    title?: IExperience['title'];
    description?: IExperience['description'];
    type?: IExperience['type'];
    category?: IExperience['category'];
    state?: IExperience['state'];
    requests?: IExperience['requests'];
    images?: IExperience['images'];
    originDate?: IExperience['originDate'];
    finalDate?: IExperience['finalDate'] | null;
    origin?: IExperience['origin'];
    destination?: IExperience['destination'] | null;
}
```
This is my updateExperience that calls that interface
export const updateExperience = async (params: IUpdateExperience) 

This is my updateExperience graphql function
```
updateExperience(
    _id: ID!
    title: String
    description: String
    type: String
    state: ExperienceState
    originDate: String
    finalDate: String
    origin: LocationInput
    destination: LocationInput
  ): Experience
```
So when I call my function in my resolver i get the error above.  I guess I could add "| null" to the IUpdateExperience, but I dont want the users to be able to update the title to null. If the title is undefined I am not updating the title.

Now in some cases I didnt have the problem above I dont know why.  Is there a procedure to avoid this errors.

**Comment by ardatan** [comment, marked as answer]
@jdc18 You can change `maybeValue` in configuration;
https://graphql-code-generator.com/docs/plugins/typescript
By default it is `T | null` so you can change it to `T` only.

**Reply by jdc18** [reply]
Is it possible to do that to only some interfaces, or do I have to do that to all the interfaces.

**Reply by dotansimha** [reply]
At the moment, it's not possible to have fine-grained configuration like that. Maybe you can share more about your use-case here?

**Comment by SzymonNadbrzezny** [comment]
> @jdc18 You can change `maybeValue` in configuration; https://graphql-code-generator.com/docs/plugins/typescript By default it is `T | null` so you can change it to `T` only.

Is there a way to do that in client-preset?
It doesn't work when using it with client-preset and website says the preset doesn't support it. https://the-guild.dev/graphql/codegen/plugins/presets/preset-client
